nengo/spa/tests/test_assoc_mem.py

```diff
@@ -35,3 +36,57 @@ def input_func(t):
     # Check to see if model builds properly. No functionality test needed
     with Simulator(m):
         pass
+
+
+def test_am_spa_keys_as_expressions(Simulator, seed, rng):
+    """Provide semantic pointer expressions as input and output keys."""
+    D = 64
+
+    voc1 = Vocabulary(D, rng=rng)
+    voc2 = Vocabulary(D, rng=rng)
+
+    voc1.parse('A+B')
+    voc2.parse('C+D+E')
+
+    in_keys = ['A', '0.5*B']
+    out_keys = ['-C+D', '0.3*E']
+
+    with nengo.spa.SPA(seed=seed) as model:
+        model.am = AssociativeMemory(input_vocab=voc1,
+                                     output_vocab=voc2,
+                                     input_keys=in_keys,
+                                     output_keys=out_keys,
+                                     threshold=0.0)
+
+        model.inp = Input(am=lambda t: 'A' if t < .5 else 'B')
+        prob = nengo.Probe(model.am.output, synapse=0.03)
+
+    with Simulator(model) as sim:
+        sim.run(1.0)
+        vec_sim = sim.data[prob]
+
+        final_vecA = vec_sim[100:490].mean(axis=0)
+        simA = np.dot(voc2.vectors, final_vecA)
```

Comment:
I think in this case, we want to use the vector norm of the vector difference (or the cosine angle) as a comparison here. The dot product is not accurate when the input vectors do not have a magnitude of 1 (`-C+D` and `0.3*E` are not guaranteed to have a magnitude of 1).